pkg/testcase/networkconnectivity.go

```diff
@@ -109,3 +110,50 @@ func testCrossNodeService(services, ports, expected []string) error {
 
 	return nil
 }
+
+func TestEndpointReadiness(cluster *factory.Cluster) {
+	//do more checks on the filesystem to ensure the certs are all created and in the correct location before this
+	commands := []string{
+		"sudo curl -sk http://127.0.0.1:10248/healthz",  //kubelet
+		"sudo curl -sk http://127.0.0.1:10249/healthz",  //kube-proxy
+		"sudo curl -sk https://127.0.0.1:10257/healthz", //kube-controller
+		"sudo curl -sk https://127.0.0.1:10258/healthz", //cloud-controller
+		"sudo curl -sk https://127.0.0.1:10259/healthz", //kube-scheduler
+		"sudo curl -sk  " + fmt.Sprintf("--cert /var/lib/rancher/%s/server/tls/client-ca.crt", cluster.Config.Product) + fmt.Sprintf(" --key  /var/lib/rancher/%s/server/tls/client-ca.key", cluster.Config.Product) + " https://127.0.0.1:6443/healthz",
+		// {Command: "sudo curl -sk http://127.0.0.1:10256/healthz"}, //SearchString: "lastUpdated" or "nodeEligible: true" //check with devs for this versus second kube-proxy port
+		// "sudo curl -sk " + fmt.Sprintf("--cert /var/lib/rancher/%s/server/tls/etcd/server-client.crt", cluster.Config.Product) + fmt.Sprintf(" --key /var/lib/rancher/%s/server/tls/etcd/server-client.key", cluster.Config.Product) + " https://127.0.0.1:2379/livez?verbose",
+	}
+	var err error
+	for _, serverIP := range cluster.ServerIPs {
+		for _, endpoint := range commands {
+			fmt.Printf("Running command %s against server %s", commands, serverIP)
+			err = assert.CheckComponentCmdNode(
+				endpoint,
+				serverIP,
+				"ok")
+		}
+	}
+	Expect(err).NotTo(HaveOccurred(), err)
+}
+
+func Testk8sAPIReady(cluster *factory.Cluster) {
+	for _, serverIP := range cluster.ServerIPs {
+		err := assert.CheckComponentCmdNode(
```

Comment:
since its a lot of commands here to be executed on the IP i would strongly recommend as performance wise inside your for loop execute it concurrently in go routines
if you need any help i would be glad to

Reply:
I've updated the PR with using go routines - I tested against a single node cluster and a 5 server 1 agent cluster with good results against all of the servers.